cli/process.ts

```diff
@@ -0,0 +1,41 @@
+import { Buffer, readLines } from "../deps/std/io.ts"
+import { readerFromStreamReader, writeAll } from "../deps/std/streams.ts"
+
+export interface DenoRunFlags {
+  reload?: string
+}
+
+export function runDeno({ reload }: DenoRunFlags) {
+  return async (pathname: string, logs: Buffer): Promise<number> => {
+    const flags = reload ? [`-r${reload === "" ? "" : `=${reload}`}`] : []
+    const process = new Deno.Command(Deno.execPath(), {
+      args: ["run", "-A", ...flags, pathname],
+      stdout: "piped",
+      stderr: "piped",
+    }).spawn()
+    return runProcess(process, logs)
+  }
+}
+
+export function runNode({}: Record<string, unknown>) {
+  return async (pathname: string, logs: Buffer): Promise<number> => {
+    const process = new Deno.Command("node", {
+      args: [pathname],
+      stdout: "piped",
+      stderr: "piped",
+    }).spawn()
+    return runProcess(process, logs)
+  }
+}
+
+async function runProcess(process: Deno.ChildProcess, logs: Buffer) {
```

Comment:
I don't love this name given that it accepts the already-run/spawned process. Any other name ideas?

Reply:
pipeLogs?

Reply:
went with collectResults